Routing.Test/RoutingTest.cs

```diff
@@ -256,6 +257,46 @@ public void ParsesMultipleRouteParams()
             }, CreateExpectedParams((NameKey, firstParam), (AgeKey, secondParam)), RegisteredRouteWithParams);
         }
 
+        [Fact]
+        public static void RoutesParameterRouteWhenLiteralRootIsAvailable()
+        {
+            const string name = "foo";
+            AssertRouteWasCalledWithParams(routeRegistry =>
+            {
+                routeRegistry
+                    .Register(HttpMethod.Get, RootRoute, FailOnRequest)
+                    .Route(HttpMethod.Get, $"/{name}", new Unit());
+            }, CreateExpectedParams((NameKey, name)), RootRouteWithParam);
+        }
+
+        [Fact]
+        public static void LiteralsArePreferredFromLeftToRight()
+        {
+            var expectedRoute = $"{RegisteredRoute}/foo/{{bar}}/{{baz}}";
+            var moreLiteralRoute = $"{RegisteredRoute}/{{foo}}/bar/baz";
+
+            AssertRouteWasCalledWithParams(routeRegistry =>
+            {
+                routeRegistry
+                    .Register(HttpMethod.Get, moreLiteralRoute, FailOnRequest)
+                    .Route(HttpMethod.Get, $"{RegisteredRoute}/foo/bar/baz", new Unit());
+            }, CreateExpectedParams(("bar", "bar"), ("baz", "baz")), expectedRoute);
+        }
+
+        [Fact]
+        public static void FirstRegisteredAmbiguousRouteIsPreferred()
+        {
+            const string name = "foo";
+            var ambiguousRoute = $"{RegisteredRoute}/{{bar}}";
+
+            AssertRouteWasCalledWithParams(routeRegistry =>
```

Comment:
Put the non-function parameters first.

Reply:
https://github.com/messerli-informatik-ag/routing/pull/6/commits/2498e174f580c386d1be3efa81eb09a9db22a977